pkg/controller/deployment/sync.go

```diff
@@ -504,6 +507,13 @@ func calculateStatus(allRSs []*apps.ReplicaSet, newRS *apps.ReplicaSet, deployme
 		UnavailableReplicas: unavailableReplicas,
 		CollisionCount:      deployment.Status.CollisionCount,
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.DeploymentPodReplacementPolicy) {
+		if terminatingReplica, err := deploymentutil.GetTerminatingReplicaCountForReplicaSets(allRSs); err != nil {
+			logger.Error(err, "failed to calculate .status.terminatingReplicas", "deployment", klog.KObj(deployment))
+		} else {
+			status.TerminatingReplicas = terminatingReplica
+		}
```

Comment:
assigning nil is not really an error condition, and not something we need to log

I think this can simplify to 
```go
status.TerminatingReplicas = deploymentutil.GetTerminatingReplicaCountForReplicaSets(allRSs)
```

and then we won't need the context / logger